webhooks/appmesh/virtualgateway_mutator_test.go

```diff
@@ -0,0 +1,216 @@
+package appmesh
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	mock_mesh "github.com/aws/aws-app-mesh-controller-for-k8s/mocks/aws-app-mesh-controller-for-k8s/pkg/mesh"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func Test_virtualGatewayMutator_defaultingAWSName(t *testing.T) {
+	type args struct {
+		vGateway *appmesh.VirtualGateway
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *appmesh.VirtualGateway
+		wantErr error
+	}{
+		{
+			name: "VirtualGateway didn't specify awsName",
+			args: args{
+				vGateway: &appmesh.VirtualGateway{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "my-ns",
+						Name:      "my-vg",
```

Comment:
I think this should also include `UID`.

Reply:
it's ok to don't have UID here per this test case's purpose :D